hedera-node/hedera-networkadmin-service/src/main/java/com/hedera/node/app/service/networkadmin/ReadableSpecialFileStore.java

```diff
@@ -22,7 +22,10 @@
 /**
  * Provides read-only methods for interacting with the underlying data storage mechanisms for
  * working with special files used in freeze transactions.
+ * <br/>
+ * @deprecated This is a temporary interface. It will be replaced by a new interface in FileService.
```

Comment:
Is this needed for anything other than the upgrade file?

Reply:
The upgrade file is the only thing that SpecialFileStore used for in freeze service. I think that it's used for other files elsewhere, but I'm not sure.

Reply:
I suspect this is only needed for the UpgradeFile, as the other "special files" are all treated entirely as normal files except for permissions checks and updates to in-memory data *after* a file update transaction.  I doubt it will ever be in FileService as it's not really part of FileService (which manages files stored in the network state, not on disk).